### PR TITLE
glob: introduce subpackage to match capture and replace using globbing patterns

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: pkg/glob/glob.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	"time"
+	"unicode/utf8"
+
+	"darvaza.org/core"
+	"github.com/dlclark/regexp2"
+	"github.com/pachyderm/ohmyglob/compiler"
+	"github.com/pachyderm/ohmyglob/syntax"
+)
+
+// this file is a slight variation of
```
Comment (karasz) [inline pkg/glob/glob.go:15]: either `slightly modified` or `light`....

File: pkg/glob/glob.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+// https://github.com/pachyderm/ohmyglob/blob/master/glob.go
+// distributed under the MIT licence.
+
+const matchTimeout = 10 * time.Second
```
Comment (karasz) [inline pkg/glob/glob.go:19]: thx ;-)

File: pkg/glob/glob.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+// Capture returns the list of sub-expressions captured by the pattern
+// against the given fixture, and also indicates if there was a match
+// at all.
+func (g *Glob) Capture(fixture string) ([]string, bool) {
+	m, err := g.r.FindStringMatch(fixture)
+	if err != nil {
+		// timeout
+		core.PanicWrap(err, "regexp2.Regexp")
+	}
+
```
Comment (karasz) [inline pkg/glob/glob.go:45]: I have a slight preference towards:
```go
if m, err:=g.r.FindStringMatch(fixture); err!=nil{
core.PanicWrap(err,"regexp2.Regexp")
}
```

**Review by karasz — CHANGES_REQUESTED** [review]

File: pkg/glob/glob.go (hunk 1 above)
Comment (karasz) [reply pkg/glob/glob.go:15]: but slight variation might work

File: pkg/glob/glob.go (hunk 3 above)
Comment (amery) [reply pkg/glob/glob.go:45]: problem there is the scope of `m`

Comment (amery) [reply pkg/glob/glob.go]: replaced with a `switch`, better?

**Comment by amery** [comment]
rebased and amended as requested

**Comment by amery** [comment]
Fixed refactoring residues

**Review by karasz — APPROVED** [review]
\o/

**Comment by amery** [comment]
`darvaza.org/sidecar@v0.2.0` tagged

**Pull request merged**